crates/infra/cli/src/toolchains/napi/compiler.rs

```diff
@@ -22,36 +22,36 @@ pub enum NapiProfile {
 pub struct NapiCompiler;
 
 impl NapiCompiler {
-    pub fn run(profile: NapiProfile) -> Result<()> {
+    pub fn run(resolver: &NapiResolver, profile: NapiProfile) -> Result<()> {
         match profile {
             NapiProfile::Debug => {
                 // Compiles the default target for local development
-                let napi_output = compile_target(&BuildTarget::Debug)?;
+                let napi_output = compile_target(resolver, &BuildTarget::Debug)?;
 
                 // Process its generated JavaScript and TypeScript files, and copy any new changes to the source folder.
-                process_generated_files(&napi_output)?;
+                process_generated_files(resolver, &napi_output)?;
 
                 // Compile the main cross-platform package, and copy the build node binary to it for debugging/testing.
-                compile_root_package(Some(&napi_output.node_binary))?;
+                compile_root_package(resolver, Some(&napi_output.node_binary))?;
             }
             NapiProfile::Release => {
                 // Compile all available targets for publishing to NPM.
-                let node_binaries = compile_all_targets()?;
+                let node_binaries = compile_all_targets(resolver)?;
 
                 // Compile the main cross-platform package, but without any binaries.
-                compile_root_package(None)?;
+                compile_root_package(resolver, None)?;
 
                 // Compile all platform-specific packages, and copy the built node binaries to them.
-                compile_platform_packages(&node_binaries)?;
+                compile_platform_packages(resolver, &node_binaries)?;
             }
         }
 
         Ok(())
     }
 }
 
-fn compile_all_targets() -> Result<Vec<PathBuf>> {
-    let targets = NapiConfig::list_all_targets()?;
+fn compile_all_targets(resolver: &NapiResolver) -> Result<Vec<PathBuf>> {
+    let targets = NapiConfig::list_all_targets(resolver)?;
```

Comment:
Not sure if we need to add/maintain the packages in `crates/testlang/outputs/npm/package/platforms`? since they are only used for release. If not, I suggest:

1. Removing the platform-specific packages from `crates/testlang/outputs/npm/package/platforms`.
2. Clearing `additional[]` in `crates/testlang/outputs/npm/package/package.json`.
3. Adding a check to `compiler.rs` for this. e.g.:

```suggestion
    let targets = NapiConfig::list_all_targets(resolver)?;

    assert_ne!(targets.len(), 0, "No release targets found. Is this a publicly released package?");
```

Reply:
Done